bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -308,6 +330,35 @@ public ObjectId(final byte[] bytes) {
     }
 
     /**
+      * Convert to bytes and put those bytes to the provided ByteBuffer.
+      * Note that the numbers are stored in big-endian order.
+      *
+      * @param buffer the ByteBuffer
+      * @throws IllegalArgumentException if the buffer is null or does not have at least 12 bytes remaining
+      */
+    public void putToByteBuffer(final ByteBuffer buffer) {
```

Comment:
Writing toByteArray in terms of putToByteBuffer will incur an allocation of a ByteBuffer that will just be thrown away (and same for the ByteBuffer.wrap in the constructor). Is that okay?